### PR TITLE
474: Add support for 3D z-sigma settings in MDU

### LINKED ISSUE #474: z-sigma layers not supported by FMModel
**Describe the bug**
Currently FMModel does not support geometry keywords for (3D) z-sigma layer models. When reading a mdu, these keywords end up in the comments, but are not read correctly. When saving the model, the information is lost. Similarly, it is not possible to initialize a FMModel with z-sigma layers, because the keywords cannot be set. Keywords are:
```
> Numtopsiguniform                    = 1                                                                   # Number of sigma layers in top of z-layer model
> Dztop                               = 5.0                                                                 # Z-layer thickness of layers above level Dztopuniabovez
> Floorlevtoplay                      = -5.0                                                                # Floor level of top layer
> Dztopuniabovez                      = -100.0                                                              # Above level Dztopuniabovez layers will have uniform Dztop, SigmaGrowthFactor below this level
> Keepzlayeringatbed                  = 2                                                                   #  bedlayerthickness = zlayerthickness at bed 0 or 1
> Dxwuimin2D                          = 0.1                                                                 # Smallest fraction dx/wu , set dx > Dxwuimin2D*wu, Default = 0.1
```

**To Reproduce**
1. test = dfm.FMModel(Path(r'p:\i1000668-tcoms\03_newModel\02_simulations\02_3D\tests\A03_coarse_53sigma\SEA_3D_3km.mdu'),path_style='unix')
2. test.save(Path('test.mdu'),recurse=False)

**Expected behavior**
Output mdu should contain the same contents.
_Related to this, but a bit of a large feature request: I think all existing keywords should be available. In practice, frequently new keywords are added and then immediately used in model applications. FMModel should be able to read this and keep it in memory, so no information gets lost. I know this is against the vision for hydrolib, but it would be good to replace the current ValueError that shows with a 'ValueWarning', so a user can continue if desired._

**Version info (please complete the following information):**
 - OS: Windows 10

### PR DESCRIPTION
**Pull request by arthurvd**
Fixes #474.

### CODE REVIEW

**Review by priscavdsluis — APPROVED** [review]

**Review by priscavdsluis — DISMISSED** [review]
@arthurvd, changes look good.
I added a comment in the reference file. Because the test is failing for MacOS and Ubuntu.

File: tests/data/reference/fm/special_3d_settings.mdu [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    wrishp_src                        = 0                                                                               # source-sinks
+    wrishp_pump                       = 0                                                                               # pumps
+    outputDir                         =                                                                                 # Output directory of map-, his-, rst-, dat- and timings-files, default: DFM_OUTPUT_<modelname>. Set to . for current dir.
+    waqOutputDir                      =                                                                                 # Output directory of Water Quality files.
+    flowGeomFile                      =                                                                                 # *_flowgeom.nc Flow geometry file in netCDF format.
+    obsFile                           = \p\i1000668-tcoms\03_newModel\01_input\05_obs\UHSLCandIHO_obs.xyn cells_obs.xyn # Points file *.xyn with observation stations with rows x, y, station name
```
[the remark this answers is not in the document]
Comment (arthurvd) [reply tests/data/reference/fm/special_3d_settings.mdu]: See latest commit

[the remark this answers is not in the document]
Comment (priscavdsluis) [reply tests/data/reference/fm/special_3d_settings.mdu]: Now it fails on Windows. We can wait for the PR, or remove the file path :)

**Comment by arthurvd** [comment]
Now ready for review approval: I merged main, which now includes #339 changes, so the last checks on Windows are now also green again.

**Review by priscavdsluis — APPROVED** [review]

**Pull request merged**